internal/resources/kubernetessecret/scope/scope.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package scope
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
+
+	secretclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+)
+
+var (
+	ScopeSchema = &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Scope for the secret having one of the valid scopes for secret: cluster or cluster_group",
+		Required:    true,
+		ForceNew:    true,
+		MaxItems:    1,
+		MinItems:    1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				ClusterKey: ClusterPolicyFullname,
+			},
+		},
+	}
+)
+
+type (
+	Scope int64
+	// ScopedFullname is a struct for all types of policy full names.
+	ScopedFullname struct {
+		Scope           Scope
+		FullnameCluster *secretclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretFullName
+	}
+)
+
+func ConstructScope(d *schema.ResourceData, name string) (scopedFullnameData *ScopedFullname) {
```

Comment:
Can we use the scope part which is already implemented?

Reply:
I don't think the same scope block was used for policies, the model for policy fullname being returned by the construct function is different than the highlighted one. 

Even if it is achieved via generics it will drastically reduce the code readability.

I prefer both the scopes to be separately defined to make both of them individually extensible. Cc: @shreyassreenivas